### PR TITLE
reconcile webhook service and deletion of webhook

### PR DESCRIPTION
**Pull request by leelavg**
remove webhook during deletion of configmap and reconcile webhook service which is deployed via bundle.

Depends on #80

### CODE REVIEW

**Comment by leelavg** [comment]
Review only last commit.

File: controllers/operatorconfigmap_controller.go [hunk 1]
```diff
@@ -131,8 +131,17 @@ func (c *OperatorConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		),
 	)
 
+	servicePredicate := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(obj client.Object) bool {
+				return obj.GetNamespace() == c.OperatorNamespace
+			},
+		),
+	)
```
Comment (nb-ohad) [inline controllers/operatorconfigmap_controller.go:140]: Why do we need this predicate if you have an ownership watch?

File: controllers/operatorconfigmap_controller.go [hunk 2]
```diff
@@ -698,10 +720,35 @@ func (c *OperatorConfigMapReconciler) reconcileSubscription() error {
...
+func (c *OperatorConfigMapReconciler) reconcileWebhookService() error {
+	svc := &corev1.Service{}
+	svc.Name = templates.WebhookServiceName
+	svc.Namespace = c.OperatorNamespace
+	err := c.createOrUpdate(svc, func() error {
+		utils.AddAnnotation(svc, "service.beta.openshift.io/serving-cert-secret-name", "webhook-cert-secret")
+		templates.WebhookService.Spec.DeepCopyInto(&svc.Spec)
+		// the service is already deployed as part of olm bundle as that'll generate certs which is required to start manager.
+		// in reconcile we are making sure that service doesn't get deleted as long as the configmap stays by owning the service.
+		return c.own(svc)
```
Comment (nb-ohad) [inline controllers/operatorconfigmap_controller.go:732]: To follow with out conventions, owning the resource should be the first thing to do inside the `CreateOrUpdate` mutator.

File: controllers/operatorconfigmap_controller.go [hunk 3]
```diff
@@ -698,10 +720,35 @@ func (c *OperatorConfigMapReconciler) reconcileSubscription() error {
...
 
+func (c *OperatorConfigMapReconciler) reconcileWebhookService() error {
+	svc := &corev1.Service{}
+	svc.Name = templates.WebhookServiceName
+	svc.Namespace = c.OperatorNamespace
+	err := c.createOrUpdate(svc, func() error {
+		utils.AddAnnotation(svc, "service.beta.openshift.io/serving-cert-secret-name", "webhook-cert-secret")
+		templates.WebhookService.Spec.DeepCopyInto(&svc.Spec)
+		// the service is already deployed as part of olm bundle as that'll generate certs which is required to start manager.
+		// in reconcile we are making sure that service doesn't get deleted as long as the configmap stays by owning the service.
```
Comment (nb-ohad) [inline controllers/operatorconfigmap_controller.go:731]: this comment line is meaningless. we all know what `own` does

File: controllers/operatorconfigmap_controller.go (hunk 1 above)
Comment (leelavg) [reply controllers/operatorconfigmap_controller.go:140]: we get reconcile events based on Kind and this controller is for a configmap and w/o this predicate we'll get an event for all services which was owned by a configmap.

File: controllers/operatorconfigmap_controller.go (hunk 2 above)
Comment (leelavg) [reply controllers/operatorconfigmap_controller.go:732]: ya, I mistook that I was doing a deep copy of entire resource, will change.

File: controllers/operatorconfigmap_controller.go (hunk 3 above)
Comment (leelavg) [reply controllers/operatorconfigmap_controller.go:731]: ack, will remove.

**Comment by leelavg** [comment]
rebased.

**Pull request merged**